blas/impl/KokkosBlas1_axpby_unification_attempt_traits.hpp

```diff
@@ -0,0 +1,1009 @@
+          std::is_same_v<T, Kokkos::complex<std::uint64_t>>);
+}
+
+// --------------------------------
+
+template <class tExecSpace, class AV, class XMV, class BV, class YMV>
+struct AxpbyUnificationAttemptTraits {
+  static constexpr bool atDevCase =
+      KokkosKernels::Impl::kk_is_gpu_exec_space<tExecSpace>();
+  static constexpr bool atHostCase = !atDevCase;
+
+  static constexpr bool Asc  = !Kokkos::is_view_v<AV>;
+  static constexpr bool Ar0  = Tr0_val<AV>();
+  static constexpr bool Ar1s = Tr1s_val<AV>();
+  static constexpr bool Ar1d = Tr1d_val<AV>();
+  static constexpr bool Avi  = Ar0 || Ar1s || Ar1d;
+
+  static constexpr bool Xr1 = Kokkos::is_view_v<XMV> && (XMV::rank == 1);
+  static constexpr bool Xr2 = Kokkos::is_view_v<XMV> && (XMV::rank == 2);
+
+  static constexpr bool Bsc  = !Kokkos::is_view_v<BV>;
+  static constexpr bool Br0  = Tr0_val<BV>();
+  static constexpr bool Br1s = Tr1s_val<BV>();
+  static constexpr bool Br1d = Tr1d_val<BV>();
+  static constexpr bool Bvi  = Br0 || Br1s || Br1d;
+
+  static constexpr bool Yr1 = Kokkos::is_view_v<YMV> && (YMV::rank == 1);
+  static constexpr bool Yr2 = Kokkos::is_view_v<YMV> && (YMV::rank == 2);
+
+  static constexpr bool xyRank1Case = Xr1 && Yr1;
+  static constexpr bool xyRank2Case = Xr2 && Yr2;
+
+  // ********************************************************************
+  // In order to better understand the lines between now and right before
+  // the constructor, assume that all constructor checks.
+  // ********************************************************************
+
+  // ********************************************************************
+  // Declare 'AtInputScalarTypeA'
+  // ********************************************************************
+  using ScalarTypeA2_atDev =
+      typename getScalarTypeFromView<AV, Avi && atDevCase>::type;
+  using ScalarTypeA1_atDev =
+      std::conditional_t<Asc && atDevCase, AV, ScalarTypeA2_atDev>;
+
+  using ScalarTypeA2_atHost =
+      typename getScalarTypeFromView<AV, Avi && atHostCase>::type;
+  using ScalarTypeA1_atHost =
+      std::conditional_t<Asc && atHostCase, AV, ScalarTypeA2_atHost>;
+
+  using AtInputScalarTypeA =
+      std::conditional_t<atHostCase,  // 'const' not removed if present
+                         ScalarTypeA1_atHost, ScalarTypeA1_atDev>;
+
+  using AtInputScalarTypeA_nonConst = typename std::conditional_t<
+      std::is_const_v<AtInputScalarTypeA>,
+      typename std::remove_const<AtInputScalarTypeA>::type, AtInputScalarTypeA>;
+
+  static constexpr bool atInputScalarTypeA_isComplex =
+      isTypeComplex<AtInputScalarTypeA_nonConst>();
+
+  // ********************************************************************
+  // Declare 'AtInputScalarTypeX'
+  // ********************************************************************
+  using AtInputScalarTypeX =
+      typename XMV::value_type;  // 'const' not removed if present
+
+  using AtInputScalarTypeX_nonConst = typename std::conditional_t<
```

Comment:
This is already defined in Kokkos::View, replace line with:
`using AtInputScalarTypeX_nonConst = typename XMV::non_const_value_type;`